github/Organization.py

```diff
@@ -890,6 +892,159 @@ def get_outside_collaborators(self, filter_=github.GithubObject.NotSet):
             url_parameters,
         )
 
+    def get_public_key(self):
+        """
+        :calls: `GET /orgs/{org}/actions/secrets/public-key <https://docs.github.com/en/rest/reference/actions#get-an-organization-public-key>`_
+        :rtype: :class:`github.PublicKey.PublicKey`
+        """
+        headers, data = self._requester.requestJsonAndCheck(
+            "GET", f"{self.url}/actions/secrets/public-key"
+        )
+        return github.PublicKey.PublicKey(
+            self._requester, headers, data, completed=True
+        )
+
+    def create_or_update_secret(
+        self, secret_name, secret_value, visibility, selected_repository_ids=None
+    ):
+        """
+        Return whether the secret has been created
+
+        :calls: `PUT /orgs/{org}/actions/secrets/{secret_name} <https://docs.github.com/en/rest/reference/actions#create-or-update-an-organization-secret>`_
+        :param secret_name: string
+        :param secret_value: string
+        :param visibility: string. "all", "private", or "selected"
+        :param selected_repository_ids: list of int. Only available when visibility is "selected"
+        :rtype: bool
+        """
+        assert isinstance(secret_name, str), secret_name
+        assert isinstance(secret_value, str), secret_value
+        assert isinstance(visibility, str), visibility
+        if visibility != "selected":
+            assert (
+                selected_repository_ids is None
+            ), "selected_repository_ids can only be used with visibility `selected`"
+        elif selected_repository_ids is not None:
+            assert isinstance(selected_repository_ids, list) and all(
+                isinstance(repo_id, int) for repo_id in selected_repository_ids
+            ), selected_repository_ids
```

Comment:
Feels like those 2 should be raise and not assert? I quickly scrolled up the code a bit and I can't find anything else than `assert isinstance`, but here we're checking the content...

Reply:
Those probably all should not be assertions in the first place, but I'm following the library already existing conventions. I changed them to value error here.